src/shared/pkcs11-util.c

```diff
@@ -913,6 +912,7 @@ int pkcs11_token_find_private_key(
 
         if (!found_class) {
                 /* Hmm, let's slightly extend the attribute list we search for */
+                static const CK_OBJECT_CLASS required_class = CKO_PRIVATE_KEY;
```

Comment:
why static?

Reply:
It's a good question ). It was declared as static when I first saw this code, and I just did not touch it.
https://github.com/systemd/systemd-stable/blob/v255-stable/src/shared/pkcs11-util.c#L679

Reply:
why not static? that pushes this into .rodata in the ELF section, which sounds fine for something you want to take a pointer of and otherwise just use as immutable data

Reply:
i prefer security properties (such as immutability) over (premature?) optimizations in corner case code paths, generally.

Hence, please go the static way